### ISSUE TITLE
2D points on "images.txt" output file do not match with the output of "cv2.projectPoints"

### ISSUE THREAD

**Issue by RonnyVelastegui2018** [colmap/colmap#2384]
**Describe the bug**
I would like to manually project the 3D to 2D coordinates using the output files of COLMAP(cameras.txt, images.txt, and point3D.txt)
The problem is that when I use the cv2.projectPoints, the 2D points do not match the output of COLMAP.

**Code**
```
cam_width=cameras[1].width 
cam_height=cameras[1].height 
cam_x_focal_length = cameras[1].params[0] 
cam_y_focal_length = cameras[1].params[1] 
cam_x_principal_point = cameras[1].params[2] 
cam_y_principal_point = cameras[1].params[3] 
dist_coeffs = np.array([cameras[1].params[4], cameras[1].params[5], cameras[1].params[6], cameras[1].params[7]]) 
intrinsic_matrix = np.array([[cam_x_focal_length, 0, cam_x_principal_point],
                         [0, cam_y_focal_length, cam_y_principal_point],
                         [0, 0, 1]])

qvec_array = images[1].qvec 
tvec_array = images[1].tvec

rMatrix = R.from_quat(qvec_array).as_matrix() 

valid_xys_array_manuallyProjected, _ = cv2.projectPoints(valid_point3D_array, 
                                rMatrix, tvec_array, 
                                intrinsic_matrix,
                                dist_coeffs) 
valid_xys_array_manuallyProjected = np.squeeze(valid_xys_array_manuallyProjected, axis=1)

plt.figure()
plt.imshow(image)
for coord in valid_xys_array_manuallyProjected: 
    plt.scatter(coord[0], coord[1], color='red') 
plt.axis('off')
plt.savefig(os.path.join(output_folder_path, "Valid2DColmapPointsManuallyProjected_"+image_name_with_ext), bbox_inches='tight', pad_inches=0) 
plt.show()
```

**Screenshot**
The projection of the 2D points generated by colmap is:
![Valid2DColmapPoints_frame_00001](https://github.com/colmap/colmap/assets/37908283/953f6c2a-e527-4f81-af69-a1c9fee4afb8)

While the manual projection using cv2.projectPoints is:
![Valid2DColmapPointsManuallyProjected_frame_00001](https://github.com/colmap/colmap/assets/37908283/23517f51-d8a8-46f9-82ee-c0f4cf3129fd)

I think that the problem maybe is related to the format output of the coordinates in "images.txt".

**Comment by shubham-monarch** [comment]
Any chance [this](https://github.com/colmap/pycolmap/issues/230#issue-2075907777) helps?

**Comment by RonnyVelastegui2018** [comment]
@shubham-monarch not really.

I think the problem here is more related to the coordinates format in "images.txt".

Maybe I am confused with the -R^t * T formula mentioned in the COLMAP [documentation](https://colmap.github.io/format.html#images-txt).

**Comment by shubham-monarch** [comment]
> @shubham-monarch not really.
> 
> I think the problem here is more related to the coordinates format in "images.txt".
> 
> Maybe I am confused with the -R^t * T formula mentioned in the COLMAP [documentation](https://colmap.github.io/format.html#images-txt).
> 
> If you have any clue about this, it would help me a lot. I've been trying to correctly project these points from 3D to 2D manually for a few days now and I'm getting desperate :(

Can you share your 3D-2D projection implementation? Also, if possible, highlight the part of the code you feel there's an issue with?

**Comment by RonnyVelastegui2018** [comment]
@shubham-monarch 

Yes. In fact, the code above is my implementation for projecting the points from 3D to 2D.

So basically,

- First, I read the camera information from the COLMAP output "camera.txt".

- Second, I go to the COLMAP files "images.txt" and "points3D.txt" and read the points in 3D and 2D generated by COLMAP.

- Finally, I use the function "cv2.projectPoints" to project the 3D points myself in order to verify if these projected points actually match the projected points generated already by COLMAP.

But as you saw in the previous two pictures my projection with "cv2.projectPoints" is totally wrong.

So, **I suspect I'm forgetting to do some kind of simple previous operation (like taking the inverse or something like that) to the camera matrix or the 3D points** before entering them into the function "cv2.projectPoints".

```
dist_coeffs = np.array([cameras[1].params[4], cameras[1].params[5], cameras[1].params[6], cameras[1].params[7]]) 
intrinsic_matrix = np.array([[cam_x_focal_length, 0, cam_x_principal_point],[0, cam_y_focal_length, cam_y_principal_point],[0, 0, 1]])

qvec_array = images[1].qvec 
tvec_array = images[1].tvec

rMatrix = R.from_quat(qvec_array).as_matrix() 

valid_xys_array_manuallyProjected, _ = **cv2.projectPoints(valid_point3D_array, rMatrix, tvec_array,  intrinsic_matrix,dist_coeffs)** 
```

**Comment by shaunpabo** [comment]
Hi, did you manage to solve the problem? I am facing this issue as well, thanks in advance

**Comment by ahojnnes** [comment]
Closing: user error; quaternion convention mismatch between COLMAP (w,x,y,z) and scipy (x,y,z,w).
